src/main/java/core/basesyntax/strategy/handler/BalanceHandler.java

```diff
@@ -0,0 +1,16 @@
+package core.basesyntax.strategy.handler;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.service.OperationHandler;
+
+public class BalanceHandler implements OperationHandler {
+    @Override
+    public void apply(Fruit fruit, int quantity) {
+        if (quantity < 0) {
+            throw new RuntimeException("Balance cannot be negative");
+        }
+        fruit.setQuantity(quantity);
+        Storage.addOrUpdateFruit(fruit.getName(), fruit);
+    }
```

Comment:
The `apply` method is directly modifying the `Storage`. To follow SOLID principles, particularly Single Responsibility and Dependency Inversion, consider creating a service that interacts with the `Storage` and use it here instead of directly accessing the `Storage` class.

Reply:
The logic was refactored